glue_exp/tools/floodfill_selection/floodfill_selection.py

```diff
@@ -0,0 +1,187 @@
+        ----------
+        data : `numpy.ndarray`
+            The dataset to use
+
+        Returns
+        -------
+        :class:`~glue.core.roi.PolygonalROI` or `None`
+            A new ROI made by the (single) floodfill that passes through the
+            mouse location (and `None` if this could not be calculated)
+        """
+        # here the xy refers to the x and y related to the order of left panel
+
+        x, y = self._event_xdata, self._event_ydata
+        slc = slc
+        return floodfill_to_roi(x, y, data, slc, data_object)
+
+
+def floodfill_to_roi(x, y, data, slc, data_object):
+    """
+    Return a PolygonalROI for the floodfill that passes through (x,y) in data
+
+    Parameters
+    ----------
+    x, y : float
+        x and y coordinate of the point
+    data : `numpy.ndarray`
+        The data
+
+    Returns
+    -------
+    :class:`~glue.core.roi.PolygonalROI`
+        A new ROI made by the (single) floodfill that passes through the
+        mouse location (and `None` if this could not be calculated)
+    """
+
+    if x is None or y is None:
+        return None
+# TODO: replace
+#     xy = point_floodfill(x, y, data)
+    # calculate the threshold and call draw visual
+    # width = event.pos[0] - self.selection_origin[0]
+    # height = event.pos[1] - self.selection_origin[1]
+    # drag_distance = math.sqrt(width**2+height**2)
+    # canvas_diag = math.sqrt(self._vispy_widget.canvas.size[0]**2
+    #                         + self._vispy_widget.canvas.size[1]**2)
+
+    # mask = self.draw_floodfill_visual(drag_distance / canvas_diag)
+    formate_data = np.asarray(data, np.float64)
+    z = slc
+    threshold = 8.
+    print('x, y, z', x, y, z)
+    # coordinate should be integers as index for array
+    mask = floodfill_scipy(formate_data, (z, int(round(y)), int(round(x))), threshold).astype(float) * 5
+
+    if mask is not None:
+        new_mask = np.reshape(mask, data.shape)
+        new_mask = np.ravel(np.transpose(new_mask))
+        print('new_mask', np.sum(new_mask))
+        mark_selected(new_mask, data, data_object)
+    # if xy is None:
+    #     return None
+
+    # p = roi.PolygonalROI(vx=xy[:, 0], vy=xy[:, 1])
+    # return p
+
+
+def mark_selected(mask, visible_data, data_object):
+    # We now make a subset state(description for selection). For scatter plots
+    # we'll want to use an ElementSubsetState, while for cubes, we'll need to
+    # change to a MaskSubsetState.
+    subset_state = PatchedElementSubsetState(visible_data, np.where(mask)[0])
+
+    # We now check what the selection mode is, and update the selection as
+    # needed (this is delegated to the correct subset mode).
+    mode = EditSubsetMode()
+    focus = visible_data[0] if len(visible_data) > 0 else None
+    # mode.update(self._data_collection, subset_state, focus_data=focus)
+    print('input for mark_selected', data_object, subset_state, focus)
+    mode.update(data_object, subset_state, focus_data=focus)
```

Comment:
I don't know how to get current data_collection so I use the `imageWidget.client.display_data' as the first input for `mode.update()`, is it the reason why there is no subset showed up? How does the `mode.update` find current subset and update it?